ImageBuilder won't use local packages as APK doesn't take relative path from `repositories` anymore

### Describe the bug

ImageBuilder has a local repo `packages`.

https://github.com/openwrt/openwrt/blob/main/target/imagebuilder/Makefile#L52

Commit 83d2d21904e0 included alpinelinux/apk-tools@460d62ee743c into APK and since then it doesn't take relative path (to `packages`) from `repositories` anymore.

CC: @Ansuel

### OpenWrt version

r28854-b74318481b

### OpenWrt release

SNAPSHOT

### OpenWrt target/subtarget

x86/64

### Device

QEMU Standard PC (Q35 + ICH9, 2009)

### Image kind

Official downloaded image

### Steps to reproduce

1. Download [ImageBuilder](https://downloads.openwrt.org/snapshots/targets/x86/64/openwrt-imagebuilder-x86-64.Linux-x86_64.tar.zst)
2. Extract it
3. Put a custom built package (`foobar` here) into `packages`
4. Build with command: `make image PROFILE=generic PACKAGES=foobar`

### Actual behaviour

```log
Building images for x86 - Generic x86/64
Packages: foobar apk-mbedtls base-files ca-bundle dnsmasq dropbear e2fsprogs firewall4 fstools grub2-bios-setup kernel kmod-amazon-ena kmod-amd-xgbe kmod-bnx2 kmod-button-hotplug kmod-drm-i915 kmod-dwmac-intel kmod-e1000 kmod-e1000e kmod-forcedeth kmod-fs-vfat kmod-igb kmod-igc kmod-ixgbe kmod-nft-offload kmod-r8169 kmod-tg3 libc libgcc libustream-mbedtls logd mkf2fs mtd netifd nftables odhcp6c odhcpd-ipv6only partx-utils ppp ppp-mod-pppoe procd-ujail uci uclient-fetch urandom-seed urngd

WARNING: /builder/openwrt-imagebuilder-x86-64.Linux-x86_64/repositories:9: unrecogized keyword: packages/packages.adb
ERROR: failed to read repositories: /builder/openwrt-imagebuilder-x86-64.Linux-x86_64/repositories: unsupported repositories file keyword
OK: 0 MiB in 0 packages
Package list missing or not up-to-date, generating it.

Building package index...

Installing packages...
WARNING: /builder/openwrt-imagebuilder-x86-64.Linux-x86_64/repositories:9: unrecogized keyword: packages/packages.adb
ERROR: failed to read repositories: /builder/openwrt-imagebuilder-x86-64.Linux-x86_64/repositories: unsupported repositories file keyword
ERROR: unable to select packages:
  foobar (no such package):
    required by: world[foobar]
make[2]: *** [Makefile:225: package_install] Error 1
make[1]: *** [Makefile:156: _call_image] Error 2
make: *** [Makefile:331: image] Error 2
```

### Expected behaviour

Image should be built with custom package included.

### Additional info

_No response_

### Diffconfig

```text

```

### Terms

- [x] I am reporting an issue for OpenWrt, not an unsupported fork.

--- a/target/imagebuilder/Makefile
+++ b/target/imagebuilder/Makefile
@@ -49,7 +49,6 @@ ifneq ($(CONFIG_USE_APK),)
 	$(call FeedSourcesAppendAPK,$(PKG_BUILD_DIR)/repositories)
 	$(VERSION_SED_SCRIPT) $(PKG_BUILD_DIR)/repositories
   endif
-	echo "packages/packages.adb" >> $(PKG_BUILD_DIR)/repositories
 
 	$(INSTALL_DATA) ./files/README.apk.md $(PKG_BUILD_DIR)/packages/README.md
 else

--- a/target/imagebuilder/files/Makefile
+++ b/target/imagebuilder/files/Makefile
@@ -99,6 +99,7 @@ OPKG:=$(call opkg,$(TARGET_DIR)) \
 export APK_KEYS:=$(TOPDIR)/keys
 APK:=$(call apk,$(TARGET_DIR)) \
 	--repositories-file $(TOPDIR)/repositories \
+	--repository $(PACKAGE_DIR)/packages.adb \
 	$(if $(CONFIG_SIGNATURE_CHECK),,--allow-untrusted) \
 	--cache-dir $(DL_DIR)
 

